applications/call_inspector/src/ci_datastore.erl

```diff
@@ -0,0 +1,272 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2014, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%
+%%%-------------------------------------------------------------------
+-module(ci_datastore).
+
+-behaviour(gen_server).
+
+-include("call_inspector.hrl").
+
+%% API
+-export([start_link/0]).
+-export([store_chunk/1]).
+-export([store_analysis/1]).
+-export([lookup_callid/1]).
+-export([callid_exists/1]).
+-export([flush/0
+         ,flush/1
+        ]).
+
+%% gen_server callbacks
+-export([init/1
+         ,handle_call/3
+         ,handle_cast/2
+         ,handle_info/2
+         ,terminate/2
+         ,code_change/3
+        ]).
+
+-record(state, {}).
+-type state() :: #state{}.
+
+-record(object, {call_id
+                 ,timestamp
+                 ,type
+                 ,value
+                }).
+-type object() :: #object{}.
+
+-define(SERVER, ?MODULE).
+-define(CI_DIR, "/tmp/2600hz-call_inspector").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Starts the server
+%%
+%% @spec start_link() -> {ok, Pid} | ignore | {error, Error}
+%% @end
+%%--------------------------------------------------------------------
+start_link() ->
+    gen_server:start_link({'local', ?SERVER}, ?MODULE, [], []).
+
+-spec store_chunk(ci_chunk:chunk()) -> 'ok'.
+store_chunk(Chunk) ->
+    'true' = ci_chunk:is_chunk(Chunk),
+    CallId = ci_chunk:call_id(Chunk),
+    gen_server:cast(?SERVER, {'store_chunk', CallId, Chunk}).
+
+-spec store_analysis(ci_analysis:analysis()) -> 'ok'.
+store_analysis(Analysis) ->
+    'true' = ci_analysis:is_analysis(Analysis),
+    CallId = ci_analysis:call_id(Analysis),
+    gen_server:cast(?SERVER, {'store_analysis', CallId, Analysis}).
+
+-spec callid_exists(ne_binary()) -> boolean().
+callid_exists(CallId) ->
+    File = make_name(CallId),
+    filelib:is_file(File).
+
+-type datum() :: {'chunk', [ci_chunk:chunk()]} |
```

Comment:
You have `'chunk`' here; but in `ci_lookup_req` you access `'chunks'` from the `Props`. Which is it??? Make up your mind already!!! :)